### PR TITLE
fix(suite): fix logic of description selection

### LINKED ISSUE #3914: Firmware update says already updated to latest firmware even when device has old firmware
**Describe the bug**

When trigger firmware update manually from Suite seeings it says already updated to latest firmware even when device has old firmware.
+ There is no message about availability of new firmware in application or during onboarding.

`Your device is already updated to latest firmware. You may reinstall the firmware if needed.`

**Steps to reproduce:**
1. Connect initialized device with 1.6.2 firmware
2. Start and finish onboarding 
3. Start firmware update manually

**Info:**
 - Suite version: desktop 21.6.2 (c0e30df7aa3c7b37e6f0a19b5201744a5de66a90)
 - Browser: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) TrezorSuite/21.6.2 Chrome/89.0.4389.69 Electron/12.0.0 Safari/537.36
 - OS: Linux x86_64
 - Screen: 2560x1440
 - Device: model T 2.4.0 regular

**Screenshots:**

Incorrect text
image

No message about firmware update 
image

### PR DESCRIPTION
**Pull request by matejkriz**
- device is already up-to-date if firmware has state 'valid' not if next available FW is the latest

fix #3914

- [ ] get back original copy

### CODE REVIEW

**Comment by goodhoko** [comment]
> device is already up-to-date if firmware has state 'valid' not if next available FW is the latest

On first and second sight this seems just wrong. What other condition than `device.fwVersion.isLatest` should be used to determine if device's FW is up-to-date? Maybe I'm just missing the meaning of `version.valid`...? o.O

**Comment by matejkriz** [comment]
> > device is already up-to-date if firmware has state 'valid' not if next available FW is the latest
> 
> On first and second sight this seems just wrong. What other condition than `device.fwVersion.isLatest` should be used to determine if device's FW is up-to-date? Maybe I'm just missing the meaning of `version.valid`...? o.O

Probably the naming is the issue here: 

- `device.firmware` ~ status of the firmware currently installed on the device (available values: 'valid' | 'outdated' | 'required' | 'unknown' | 'none')
- `device.firmwareRelease` ~ available firmware to update to (it is whatever returns getInfo() method from trezor-rollout)

File: packages/suite/src/components/firmware/FirmwareInitial.tsx [hunk 1]
```diff
@@ -89,11 +97,11 @@ const FirmwareInitial = ({ cachedDevice, setCachedDevice, standaloneFwUpdate }:
...
                 <Translation
-                    id={getDescription(
-                        device.firmware === 'required',
-                        !!standaloneFwUpdate,
-                        !!device.firmwareRelease?.isLatest,
-                    )}
+                    id={getDescription({
+                        required: device.firmware === 'required',
+                        standaloneFwUpdate,
+                        reinstall: device.firmware === 'valid',
```
Comment (goodhoko) [inline packages/suite/src/components/firmware/FirmwareInitial.tsx:111]: Could we add a comment explaning this in similar way as you did [here](https://github.com/trezor/trezor-suite/pull/3961#issuecomment-871417500)?

**Review by goodhoko — APPROVED** [review]
LGTM with one rather minor remark. Feel fry to resolve it yourself and merge this.

File: packages/suite/src/components/firmware/FirmwareInitial.tsx (hunk 1 above)
Comment (matejkriz) [reply packages/suite/src/components/firmware/FirmwareInitial.tsx:111]: done, thanks

**Pull request merged**